test/Garnet.test/RespListTests.cs

```diff
@@ -97,24 +96,66 @@ public void MultiLPUSHAndLTRIM()
 
             //all elements remain
             db.ListTrim(key, 0, -1);
-            nLen1 = db.ListLength(key);
-            Assert.AreEqual(nLen1, 5);
+            nLen = db.ListLength(key);
+            Assert.AreEqual(5, nLen);
 
             result = db.Execute("MEMORY", "USAGE", key);
             actualValue = ResultType.Integer == result.Resp2Type ? Int32.Parse(result.ToString()) : -1;
             expectedResponse = 504;
             Assert.AreEqual(expectedResponse, actualValue);
 
             db.ListTrim(key, 0, -3);
-            nLen1 = db.ListLength(key);
-            Assert.AreEqual(3, nLen1);
+            nLen = db.ListLength(key);
+            Assert.AreEqual(3, nLen);
 
             var vals = db.ListRange(key, 0, -1);
             Assert.AreEqual("val_8", vals[0].ToString());
             Assert.AreEqual("val_7", vals[1].ToString());
             Assert.AreEqual("val_6", vals[2].ToString());
         }
 
+        private static object[] LTrimTestCases = {
+            new object[] {0, 0, new[] {0} },
+            new object[] {-2, -1, new[] {8, 9} },
+            new object[] {-2, -2, new[] {8} },
+            new object[] {3, 5, new[] {3, 4, 5} },
+            new object[] {-12, 0, new[] {0} },
+            new object[] {-12, 2, new[] {0, 1, 2} },
+            new object[] {-12, -7, new[] {0, 1, 2, 3} },
+            new object[] {-15, -11, Array.Empty<int>() },
+            new object[] {8, 8, new[] {8} },
+            new object[] {8, 12, new[] {8, 9} },
+            new object[] {9, 12, new[] {9} },
+            new object[] {10, 12, Array.Empty<int>() },
+        };
```

Comment:
Add a start > end case (code handles it) and per docs the key should be removed in that case or if start > end of list